### PR TITLE
Add support for Poetry 1.6

### PR DESCRIPTION
**Pull request by waketzheng**
1. Add support for Poetry 1.6;
2. Upgrade dev dependencies;
3. Drop support for python<3.8.1 because of flake8 requires.

This version of the plugin supports Poetry
>= 1.3, < 1.7.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "poetry-plugin-pypi-mirror"
-version = "0.3.2"
```
Comment (JacobHenner) [inline pyproject.toml]: Could you leave this unincremented for now? I'll increment the version at release time.

File: src/poetry_plugin_pypi_mirror/plugins.py [hunk 2]
```diff
@@ -5,7 +5,12 @@
 from cleo.io.io import IO
 from poetry.config.config import Config
 from poetry.core.packages.package import Package
-from poetry.core.semver.version import Version
+
+try:
+    from poetry.core.constraints.version import Version
+except ImportError:
+    from poetry.core.semver.version import Version  # Compare with poetry<1.6
+
```
Comment (JacobHenner) [inline src/poetry_plugin_pypi_mirror/plugins.py]: This change appears to have been made for poetry-core 1.3.0, approximately 1y ago: https://github.com/python-poetry/poetry-core/pull/482

File: pyproject.toml [hunk 3]
```diff
@@ -10,19 +10,19 @@ homepage = "https://github.com/arcesium/poetry-plugin-pypi-mirror"
 repository = "https://github.com/arcesium/poetry-plugin-pypi-mirror"
 
 [tool.poetry.dependencies]
-python = "^3.7"
```
Comment (JacobHenner) [inline pyproject.toml:13]: I don't think this comment is strictly necessary. Python 3.7 is EOL, supporting >= 3.8 only does not need further justification.

**Review by JacobHenner — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Could you edit your commit(s) to include a DCO signature? I've just added instructions to the project's README: https://github.com/arcesium/poetry-plugin-pypi-mirror#sign-your-work

File: src/poetry_plugin_pypi_mirror/plugins.py (hunk 2 above)
Comment (waketzheng) [reply src/poetry_plugin_pypi_mirror/plugins.py]: Great! Simple is better that complex~

**Comment by waketzheng** [comment]
> Thank you for your contribution!
> 
> Could you edit your commit(s) to include a DCO signature? I've just added instructions to the project's README: https://github.com/arcesium/poetry-plugin-pypi-mirror#sign-your-work

Should I sign the old commit?

**Comment by JacobHenner** [comment]
> > Thank you for your contribution!
> > Could you edit your commit(s) to include a DCO signature? I've just added instructions to the project's README: https://github.com/arcesium/poetry-plugin-pypi-mirror#sign-your-work
> 
> Should I sign the old commit?

Yes please. You can rebase your two commits into a single commit, and sign that commit. Then we should be good to go.

**Comment by waketzheng** [comment]
@JacobHenner done.

**Review by JacobHenner — APPROVED** [review]

**Pull request merged**